src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerIndexServiceTest.kt

```diff
@@ -31,21 +43,73 @@ class PrisonerIndexServiceTest {
     indexProperties
   )
 
-  @Test
-  fun `indexingComplete - clear messages if index build is complete`() {
-    whenever(indexStatusService.markRebuildComplete()).thenReturn(true)
+  @Nested
+  inner class IndexingComplete {
 
-    prisonerIndexService.indexingComplete(true)
+    @Test
+    fun `clear messages if index build is complete`() {
+      whenever(indexStatusService.markRebuildComplete()).thenReturn(true)
 
-    verify(indexQueueService).clearAllMessages()
+      prisonerIndexService.indexingComplete(true)
+
+      verify(indexQueueService).clearAllMessages()
+    }
+
+    @Test
+    fun `do not clear messages if index build is not complete`() {
+      whenever(indexStatusService.markRebuildComplete()).thenReturn(false)
+
+      prisonerIndexService.indexingComplete(true)
+
+      verifyZeroInteractions(indexQueueService)
+    }
   }
 
-  @Test
-  fun `indexingComplete - do not clear messages if index build is not complete`() {
-    whenever(indexStatusService.markRebuildComplete()).thenReturn(false)
+  @Nested
+  inner class CheckExistsAndReset {
+
+    private val restClient = mock<RestClient>()
+    private val elasticsearchOperations = mock<ElasticsearchOperations>()
+    private val indexOperations = mock<IndexOperations>()
+
+    @BeforeEach
+    fun `mock ES clients`() {
+      whenever(searchClient.lowLevelClient()).thenReturn(restClient)
+      whenever(searchClient.elasticsearchOperations()).thenReturn(elasticsearchOperations)
+      whenever(elasticsearchOperations.indexOps(any<IndexCoordinates>())).thenReturn(indexOperations)
+    }
+
+    @Test
+    fun `waits for index to be gone before trying to recreate it`() {
```

Comment:
Not over the moon about the horrible situation with mocking the elasticsearch clients in a unit test.  

Normally I'd test something like this with an integration test, but can't think of a nice way to faithfully reproduce the issue (the delete index request takes a long time).

We could move the elasticsearch clients into another class to hide the complexity, but as the client is used directly everywhere this will mean a big refactor.

So I've gone for the horrible elasticsearch client mocking.  It's only this one test so... :man_shrugging: